### PR TITLE
Add file level support for CDL objects

### PR DESCRIPTION
**Pull request by mejackreed**
## Why was this change made?
To enable file level support needed for Stacks + Purl

## How was this change tested?
Using tests!

### CODE REVIEW

File: spec/rights_auth_spec.rb [hunk 1]
```diff
@@ -39,6 +39,21 @@
     XML
   end
 
+  let(:file_cdl_xml) do
+    <<-XML
+      <rightsMetadata>
+        <access type="read">
+          <file>def456.jp2</file>
```
Comment (jcoyne) [inline spec/rights_auth_spec.rb:46]: @peetucket we need a way to represent this in the Cocina model

**Comment by peetucket** [comment]
If i'm understand this correctly, this is layering file level support on top of the draft PR I started with object level support?

**Comment by peetucket** [comment]
so a CDL type object is a new read flavor, distinct (at an object level) from world or stanford only...and then on top that, we may have specific files that are CDL and others that are not readable?

**Comment by mejackreed** [comment]
CDL is a object type similar to world or stanford only. However, for files that are part of an object, we need the ability to check a file for its type. I don't envision only specific parts of an object being "cdl", but I added that as well for consistency with how our rights metadata is modeled elsewhere. 

At the end of the day we need a useful `#cdl_rights_for_file?(file_name)` method that can give us a determination on whether or not the file is "cdl".

The main goal of this was consistency with our other rights implementations. If we need to revisit how this is done, happy to help w/ that.

**Comment by peetucket** [comment]
ok, thanks for the clarification -- i also asked in the #cdl channel because my understanding was similar, that CDL is an object level read type, alongside world, stanford, etc. and not an attribute on top of those... the distinction affects how its mapped into and out of fedora

File: lib/dor/rights_auth.rb [hunk 2]
```diff
@@ -170,6 +170,12 @@ def stanford_only_downloadable_file?(file_name)
       @file[file_name].group[:stanford].value && (stanford_rule.nil? || stanford_rule != NO_DOWNLOAD_RULE)
     end
 
+    def cdl_rights_for_file?(file_name)
```
Comment (cbeer) [inline lib/dor/rights_auth.rb:173]:
```suggestion
    def cdl_rights_for_file(file_name)
```

**Comment by jcoyne** [comment]
@mejackreed we should be able to combine CDL with other access though, right?  LIke you can see this with a location based access or with CDL.

**Comment by mejackreed** [comment]
@jcoyne do you need more here to proceed? or is what was discussed in #cdl sufficient?

**Comment by jcoyne** [comment]
@mejackreed I don't think I need anything else

**Pull request merged**